### DISCUSSION TITLE
Adding kube-scoring / preprocessing to helmreleases

### DISCUSSION THREAD

**Discussion by funkypenguin** [fluxcd/flux2#510] [Proposals] [answered]
Hi again!

Another question came up during our trial of Fluxv2...

Our current workflow (which we're trying to improve with GitOps) uses ansible playbooks to apply helm charts to our clusters. The playbook first dumps the chart in manifest form using `helm template`, and runs the resulting manifests through [kube-score](https://github.com/zegl/kube-score), for static analysis. 

This process has saved my 🥓 several times, when I've either misconfigured a chart's values, or forgotten to enable securityContexts, probes, etc.

Could I achieve same with flux? If the answer is currently "no", then would the helm controller be the appropriate place to propose this?

Cheers!
D

**Comment by stefanprodan** [comment, marked as answer]
Static analysis should be run in CI on every pull request, so you would validate the yamls before they are merged and applied on the cluster by Flux. We have no plans on allowing users to replace CI by running arbitrary binaries inside the cluster.
